### PR TITLE
Use debounce instead of throttleLast to avoid constant searching

### PR DESCRIPTION
**Pull request by apoi**
Debounce restarts timer on each change, whereas throttleLast is sampling on fixed timer. The sampling causes search to be restarted constantly, which quickly uses up our API quota when testing.

### CODE REVIEW

File: app/src/main/java/io/reark/rxgithubapp/viewmodels/RepositoriesViewModel.java [hunk 1]
```diff
@@ -103,7 +103,7 @@ public void subscribeToDataStoreInternal(@NonNull CompositeSubscription composit
         ConnectableObservable<DataStreamNotification<GitHubRepositorySearch>> repositorySearchSource =
                 searchString
                           .filter((string) -> string.length() > 2)
-                          .throttleLast(500, TimeUnit.MILLISECONDS)
+                          .debounce(1000, TimeUnit.MILLISECONDS)
```
Comment (jaggernod) [inline app/src/main/java/io/reark/rxgithubapp/viewmodels/RepositoriesViewModel.java]: Can you change to use a constant instead of magic number?

Comment (apoi) [inline app/src/main/java/io/reark/rxgithubapp/viewmodels/RepositoriesViewModel.java]: Done

**Comment by jaggernod** [comment]
Looks good

**Comment by jaggernod** [comment]
Nice :) Merge

**Pull request merged**